tests/Feature/Api/LoginTest.php

```diff
@@ -0,0 +1,120 @@
+<?php
+namespace Tests\Feature\Api;
+
+use Tests\TestCase;
+use Illuminate\Foundation\Testing\WithFaker;
+use Illuminate\Foundation\Testing\DatabaseMigrations;
+use Illuminate\Foundation\Testing\RefreshDatabase;
+use Illuminate\Support\Facades\Artisan;
+use App\Models\User;
+
+class LoginTest extends TestCase
+{
+    use DatabaseMigrations;
+
+    /**
+    * Set up database
+    *
+    * @return void
+    */
+    public function setUp()
+    {
+        parent::setUp();
+    }
+
+    /**
+    * Return structure of json.
+    *
+    * @return array
+    */
+    public function jsonStructureLoginSuccess()
+    {
+        return [
+          'result' => [
+              'token',
+              'user' => [
+                  'id',
+                  'username',
+                  'email',
+                  'old_password',
+                  'is_active',
+                  'role',
+                  'last_logined_at',
+                  'created_at',
+                  'updated_at',
+                  'deleted_at',
+                  'user_info' => [
+                      'id',
+                      'user_id',
+                      'full_name',
+                      'avatar',
+                      'gender',
+                      'dob',
+                      'address',
+                      'phone',
+                      'identity_card',
+                      'created_at',
+                      'updated_at'
+                  ]
+              ]
+          ],
+          'code'
+        ];
+    }
+
+    /**
+     * Test structure of json response.
+     *
+     * @return void
+     */
+    public function testJsonLoginSuccess()
+    {
+        $body = [
+            'email' => $this->user->email,
+            'password' => '12345'
```

Comment:
sao ko lấy `$this->user->password` lun em, để vại sao biết `$user` có pass `12345`?

Reply:
em nghĩ ở đây `$this->user->password` sẽ trả về password bị encrypt rồi ạ. nên em để pass mặc định cho user :(

Reply:
ok